src/ORM/DatabaseAdmin.php

```diff
@@ -460,7 +461,8 @@ protected function getClassNameRemappingFields()
         foreach ($dataClasses as $className) {
             $fieldSpecs = $schema->fieldSpecs($className);
             foreach ($fieldSpecs as $fieldName => $fieldSpec) {
-                if (Injector::inst()->create($fieldSpec, 'Dummy') instanceof DBClassName) {
+                $dummy = Injector::inst()->create($fieldSpec, 'Dummy');
+                if ($dummy instanceof DBClassName || $dummy instanceof DBClassNameVarchar) {
```

Comment:
This is the only logic I found that specifically care if a field is a DBClassName ... and this code is only normally called by DatabaseAdmin::migrateClassNames() which uses the config `classname_value_remapping` - which is only used when migrating sites from Silverstripe 3, so it's largely irrelevant.

Reply:
> which is only used when migrating sites from Silverstripe 3

It's useful for migrating data at any time in a project's lifetime - we should make sure this functionality is still working.